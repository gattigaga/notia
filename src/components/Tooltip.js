import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  width: auto;
  padding: 4px 8px;
  border-radius: 2px;
  background: #444;
  white-space: nowrap;
  position: absolute;
  top: 70%;
  opacity: 0;
  transition: all 0.3s;
`;

const Text = styled.span`
  color: white;
  font-family: Roboto;
  font-size: 10px;
`;

const Tooltip = ({ text, className }) => (
  <Container className={className}>
    <Text>{text}</Text>
  </Container>
);

Tooltip.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string
};

export default Tooltip;
