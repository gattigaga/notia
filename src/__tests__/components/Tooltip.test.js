import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Tooltip from "components/Tooltip";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Tooltip text="New List" />);
});

describe("Tooltip", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
