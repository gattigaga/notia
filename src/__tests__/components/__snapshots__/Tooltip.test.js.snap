// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<styled.div>
  <styled.span>
    New List
  </styled.span>
</styled.div>
`;
